pkgs/development/libraries/arcan/default.nix

```diff
@@ -0,0 +1,185 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, cmake
+, pkg-config
+, xz
+, libuvc
+, libffi
+, libX11
+, libxcb
+, libXcomposite
+, libXfixes
+, xcbutil
+, xcbutilwm
+, libXau
+, libXdmcp
+, openal
+, freetype
+, sqlite
+, libGL
+, libdrm
+, libxkbcommon
+, mesa
+, valgrind
+, luajit
+, libusb1
+, harfbuzzFull
+, SDL2
+
+# For generating API documentation
+, ruby
+
+#
+, withVLC ? true
+, libvlc
+
+#
+, withFFmpeg ? true
+, ffmpeg-full
+
+#
+, withWayland ? true
+, wayland
+, wayland-protocols
+
+#
+, withEspeak ? true
+, espeak-classic
+
+#
+, withLibmagic ? true
+, file
+
+#
+, withTesseractAndLeptonica ? true
+, tesseract
+, leptonica
+
+#
+, withVNC ? true
+, libvncserver
+
+# Output debug information
+, debugging ? false
+}:
+
+let
+  name = "arcan";
+
+  srcs = {
+    arcan = fetchFromGitHub {
+      owner = "letoram";
+      repo = name;
+      rev = "ffc266523bc89def40666cceb17be5707994d1c0";
+      sha256 = "sha256-bfLGjnnxX+sHnDn5eGYKqRR51Gn9al9GEqw9qF63j38=";
+    };
+
+    openal = fetchFromGitHub {
+      owner = "letoram";
+      repo = "openal";
+      rev = "1c7302c580964fee9ee9e1d89ff56d24f934bdef"; # master
+      sha256 = "0dcxcnqjkyyqdr2yk84mprvkncy5g172kfs6vc4zrkklsbkr8yi2";
+    };
+  };
+in stdenv.mkDerivation rec {
+  pname = name;
+
+  version = "0.6.1pre1+unstable=2021-05-31";
+
+  src = srcs.arcan;
+
+  postUnpack = ''
+    mkdir -p $sourceRoot/external/git/openal
+    pushd $sourceRoot/external/git/
+    cp --recursive --verbose ${srcs.openal}/* openal/
+    chmod --recursive 777 openal/
+    popd
+
+    # Generate manpages
+    pushd $sourceRoot/doc/
+    mkdir mantmp
+    ruby docgen.rb mangen
+    popd
+  '';
+
+  postPatch = ''
+    substituteInPlace ./src/platform/posix/paths.c \
+      --replace "/usr/bin" "$out/bin" \
+      --replace "/usr/share" "$out/share"
+
+    substituteInPlace ./src/CMakeLists.txt \
+      --replace "SETUID" ""
+  '';
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+    ruby
+  ];
+
+  buildInputs = [
+    openal
+    freetype
+    sqlite
+    libuvc
+    libGL
+    SDL2
+    libdrm
+    libxkbcommon
+    mesa
+    valgrind
+    luajit
+    libusb1
+    libvlc
+    xz
+    libffi
+    libX11
+    libxcb
+    libXcomposite
+    libXfixes
+    xcbutil
+    xcbutilwm
+    libXau
+    libXdmcp
+    harfbuzzFull
+  ] ++ lib.optionals withFFmpeg [
+    ffmpeg-full
+  ] ++ lib.optionals withWayland [
+    wayland
+    wayland-protocols
+  ] ++ lib.optionals withEspeak [
+    espeak-classic
+  ] ++ lib.optionals withLibmagic [
+    file
+  ] ++ lib.optionals withTesseractAndLeptonica [
+    tesseract
+    leptonica
+  ] ++ lib.optionals withVNC [
+    libvncserver
+  ];
+
+  cmakeFlags = [
+    "-DBUILD_PRESET=everything"
+    "-DISTR_TAG=Nixpkgs"
+    "-DENGINE_BUILDTAG=${version}+Nixpkgs"
```

Comment:
```suggestion
    "-DENGINE_BUILDTAG=${version}"
```

Reply:
Arcan's developer wanted some form of information with commit number indicating that the user use Nix. Make it easier for him to debug.